forum/communications/tests.py

```diff
@@ -1,3 +1,273 @@
+        )
+        self.assertEqual(response.status_code, 201)
+
+    def test_send_message(self):
+        room = Room.objects.create()
+        room.online.set([self.user1, self.user2])
+        self.client.force_authenticate(user=self.user1)
+        response = self.client.post(
+            "/api/v1/communications/messages/",
+            {"room": room.id, "user": self.user1, "content": "Hello!"},
+        )
+        self.assertEqual(response.status_code, 201)
+
+    def test_message_history(self):
+        room = Room.objects.create()
+        room.online.set([self.user1, self.user2])
+        Message.objects.create(room=room, user=self.user1, content="Hello!")
+        self.client.force_authenticate(user=self.user2)
+        response = self.client.get(
+            f"/api/v1/communications/conversations/{room.id}/messages/"
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data["results"]), 1)
+
+    def test_create_room_with_invalid_participants(self):
+        """Test creating a room with invalid participant data."""
+        self.client.force_authenticate(user=self.user1)
+        response = self.client.post(
+            "/api/v1/communications/conversations/", {"online": [99999, 88888]}
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("online", response.data)
+
+    def test_delete_non_existent_room(self):
+        """Test deleting a non-existent room."""
+        self.client.force_authenticate(user=self.user1)
+        response = self.client.delete("/api/v1/communications/conversations/999/")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+
+class PaginationTest(APITestCase):
+    def setUp(self):
+        self.user1 = User.objects.create_user(
+            email="user1@example.com", password="password"
+        )
+        self.user2 = User.objects.create_user(
+            email="user2@example.com", password="password"
+        )
+        self.room = Room.objects.create(name="Test Room")
+        self.room.online.set([self.user1, self.user2])
+        response = self.client.post(
+            "/auth/jwt/create/",
+            {"email": "user1@example.com", "password": "password"},
+        )
+        self.token = response.data["access"]
+
+        for i in range(11):
+            Message.objects.create(
+                content=f"Message {i+1}", user=self.user1, room=self.room
+            )
+
+    def test_pagination(self):
+
+        response = self.client.get(
+            f"/api/v1/communications/conversations/{self.room.id}/messages/",
+            HTTP_AUTHORIZATION=f"Bearer {self.token}",
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertIn("results", response.data)
+        self.assertEqual(len(response.data["results"]), 10)
+        self.assertIn("next", response.data)
+        self.assertIsNotNone(response.data["next"])
```

Comment:
Additional edge cases and assertions can make tests more robust, e.g.:
 - Validate behavior when there are fewer than 10 messages.
 - Validate the behavior for the last page (no `next` link).
 - Validate invalid page numbers (proper handling of non-existent pages)

Reply:
Done